src/msg/async/Stack.cc

```diff
@@ -37,9 +37,11 @@
 std::function<void ()> NetworkStack::add_thread(Worker* w)
 {
   return [this, w]() {
-      char tp_name[16];
-      sprintf(tp_name, "msgr-worker-%u", w->id);
-      ceph_pthread_setname(pthread_self(), tp_name);
+      if (type != "dpdk") {
+	 char tp_name[16];
+	 sprintf(tp_name,"msgr-worker-%u",w->id);
+	 ceph_pthread_setname(pthread_self(),tp_name);
+      }
```

Comment:
instead of connecting the behavior with member variable and a "magic number" (a string literal here actually), i'd suggest introduce a virtual method to `NetworkStack` class and let the derived class to signify it its thread should be renamed using `ceph_pthread_setname()` by overriding it.